src/Microsoft.Azure.WebJobs.Host/Singleton/BlobLeaseDistributedLockManager.cs

```diff
@@ -0,0 +1,404 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Executors;
+using Microsoft.Azure.WebJobs.Host.Storage;
+using Microsoft.Azure.WebJobs.Host.Storage.Blob;
+using Microsoft.Azure.WebJobs.Host.Timers;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    // Provides a blob-leased based implementation 
+    internal class BlobLeaseDistributedLockManager : IDistributedLockManager
+    {
+        internal const string FunctionInstanceMetadataKey = "FunctionInstance";
```

Comment:
The nomenclature is confusing. This is not functions specific, it's the Function instance id (which is a core SDK concept). 

---
In reply to: [121542160](https://github.com/Azure/azure-webjobs-sdk/pull/1184#discussion_r121542160) [](ancestors = 121542160)